### PR TITLE
executor: Provide the column_name, index_name in TIFLASH_INDEXES

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/55693

Problem Summary:

We support building multiple vector indexes on the same column, the `COLUMN_NAME` in `TIFLASH_INDEXES` is insufficient

### What changed and how does it work?

* Add `INDEX_NAME` in `TIFLASH_INDEXES`
* Fetch the TableInfo from infoschema and fill the `COLUMN_NAME` and `INDEX_NAME`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```sql
CREATE TABLE `t` (  `v` vector(5) DEFAULT NULL,  `v2` vector(3) DEFAULT NULL);
alter table t set tiflash replica 1;
ALTER TABLE t ADD VECTOR INDEX idx_v_l2 ((VEC_L2_DISTANCE(v))) USING HNSW;
ALTER TABLE t ADD VECTOR INDEX idx_v_cos ((VEC_COSINE_DISTANCE(v))) USING HNSW;
select * from information_schema.tiflash_indexes;
+---------------+------------+----------+-------------+------------+-----------+----------+------------+---------------------+-------------------------+--------------------+------------------------+---------------+------------------+
| TIDB_DATABASE | TIDB_TABLE | TABLE_ID | COLUMN_NAME | INDEX_NAME | COLUMN_ID | INDEX_ID | INDEX_KIND | ROWS_STABLE_INDEXED | ROWS_STABLE_NOT_INDEXED | ROWS_DELTA_INDEXED | ROWS_DELTA_NOT_INDEXED | ERROR_MESSAGE | TIFLASH_INSTANCE |
+---------------+------------+----------+-------------+------------+-----------+----------+------------+---------------------+-------------------------+--------------------+------------------------+---------------+------------------+
| test          | t          | 7401     | v           | idx_v_l2   | 1         | 1        | HNSW       | 0                   | 0                       | 0                  | 0                      |               | 10.2.12.79:9540  |
| test          | t          | 7401     | v           | idx_v_cos  | 1         | 2        | HNSW       | 0                   | 0                       | 0                  | 0                      |               | 10.2.12.79:9540  |
+---------------+------------+----------+-------------+------------+-----------+----------+------------+---------------------+-------------------------+--------------------+------------------------+---------------+------------------+
```
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/infoschema_reader.go [hunk 1]
```diff
@@ -3498,6 +3499,37 @@ func (e *TiFlashSystemTableRetriever) dataForTiFlashSystemTables(ctx context.Con
...
+		if e.table.Name.L == "tiflash_indexes" {
+			var logicalTableID = outputRow[outputColIndexMap["table_id"]].GetInt64()
+			if !outputRow[outputColIndexMap["belonging_table_id"]].IsNull() {
+				// Old TiFlash versions may not have this column. In this case we will try to get by the "table_id"
+				belongingTableID := outputRow[outputColIndexMap["belonging_table_id"]].GetInt64()
+				if belongingTableID != -1 && belongingTableID != 0 {
+					logicalTableID = belongingTableID
+				}
+			}
+			if table, ok := is.TableByID(context.Background(), logicalTableID); ok {
```
Comment (tangenta) [inline pkg/executor/infoschema_reader.go:3513]:
```suggestion
			if table, ok := is.TableByID(ctx, logicalTableID); ok {
```

Comment (JaySon-Huang) [reply pkg/executor/infoschema_reader.go]: updated, PTAL again

**Review by tangenta — APPROVED** [review]

File: pkg/executor/infoschema_reader.go [hunk 2]
```diff
@@ -3498,6 +3499,37 @@ func (e *TiFlashSystemTableRetriever) dataForTiFlashSystemTables(ctx context.Con
...
+				tableInfo := table.Meta()
+				getInt64DatumVal := func(datum_name string, default_val int64) int64 {
+					datum := outputRow[outputColIndexMap[datum_name]]
+					if !datum.IsNull() {
+						return datum.GetInt64()
+					}
+					return default_val
+				}
+				// set column_name
+				columnID := getInt64DatumVal("column_id", 0)
```
Comment (zimulala) [inline pkg/executor/infoschema_reader.go:3523]: Could we remove `column_id` in this repo?

Comment (JaySon-Huang) [reply pkg/executor/infoschema_reader.go:3523]: Removing `column_id` will make it hard to fetch the "column_id" from tiflash's `system.dt_local_index` table, and then it make more difficult to identify which column_name the index is built on. According to the previous discussion with @breezewish , we'd better provide the column_name for usability. So I'd like to keep the `column_id` field in the `information_schema.tiflash_indexes` table.

**Pull request merged**